DSharpPlus/Entities/Guild/DiscordGuild.cs

```diff
@@ -924,6 +924,22 @@ public async Task<IReadOnlyList<DiscordRole>> ModifyRolePositionsAsync(IDictiona
         this._roles = new(returnedRoles.Select(x => new KeyValuePair<ulong, DiscordRole>(x.Id, x)));
         return returnedRoles;
     }
+    
+    /// <summary>
+    /// Removes a specified member from this guild.
+    /// </summary>
+    /// <param name="member">Member to remove.</param>
+    /// <param name="reason">Reason for audit logs.</param>
+    public async Task RemoveMemberAsync(DiscordMember member, string reason = null)
+        => await this.Discord.ApiClient.RemoveGuildMemberAsync(this.Id, member.Id, reason);
+
+    /// <summary>
+    /// Removes a specified member by ID.
+    /// </summary>
+    /// <param name="user_id">ID of the user to remove.</param>
+    /// <param name="reason">Reason for audit logs.</param>
+    public async Task RemoveMemberAsync(ulong user_id, string reason = null)
```

Comment:
userId, `string? reason`, though i'm not sure whether we would want this overload

Reply:
i don't understand why, we have DiscordGuild.BanMember, I think this addition completes the rest

Reply:
you can ban by ID because you can ban people that aren't in your guild, but you can't kick users that aren't in your guild. that said, the inconsistency of cache and having DiscordUser in slash commands is a sufficiently compelling point to expose it anyway